src/create-reducer.test.js

```diff
@@ -1,3 +1,26 @@
+import createReducer from './create-reducer';
+import when from './when';
+
+const reducer = createReducer(
+	[when('ACTION', state => ({...state, correct: true}))],
+	{},
```

Comment:
Hey @mariosant 

This seems like a great job. 
If I may suggest something, it would be to consider making this test a bit more expressive. So, instead of using an empty object `{}` you assign that to a variable and use that one.

Here's an example:

```js
const initialState = {};

const reducer = createReducer(
	[when('ACTION', state => ({...state, correct: true}))],
	initialState,
 ); 

// And later on your test
describe('createReducer', () => {
	test('will return initial state when state is undefined', () => {
		expect(reducer(undefined, {type: 'ACTION', payload: 'test'})).toEqual(initialState);
	});
```

The only reason I am suggesting this is that it took me a few seconds that the empty object you are referring to inside the test is the same object that you pass to `createReducer`. Maybe something like is even better:

```js
const initialState = {initial: 'state'};
```

What do you think?

Reply:
Thanks @george-aidonidis! Happy to make this test a bit easier to comprehend.
What if `initialState` is a string like 'initial-state`, instead? A reducer could also return this instead of an object as well.

Reply:
Done as mentioned above.